gpt_index/callbacks/base.py

```diff
@@ -0,0 +1,85 @@
+import uuid
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional
+
+from gpt_index.callbacks.schema import CBEventType
+
+
+class BaseCallbackHandler(ABC):
+    """Base callback handler that can be used to track event starts and ends."""
+
+    def __init__(
+        self,
+        event_starts_to_ignore: List[CBEventType],
+        event_ends_to_ignore: List[CBEventType],
+    ) -> None:
+        """Initialize the base callback handler."""
+        self.event_starts_to_ignore = tuple(event_starts_to_ignore)
+        self.event_ends_to_ignore = tuple(event_ends_to_ignore)
+
+    @abstractmethod
+    def on_event_start(
+        self,
+        event_type: CBEventType,
+        payload: Optional[Dict[str, Any]] = None,
+        event_id: str = "",
+        **kwargs: Any
+    ) -> str:
+        """Run when an event starts and return id of event."""
+
+    @abstractmethod
+    def on_event_end(
+        self,
+        event_type: CBEventType,
+        payload: Optional[Dict[str, Any]] = None,
+        event_id: str = "",
+        **kwargs: Any
+    ) -> None:
+        """Run when an event ends."""
+
+
+class CallbackManager(BaseCallbackHandler, ABC):

```

Comment:
curious: what's the conceptual difference between llama logger and the callback manager? Could/should they be unified?

Reply:
Yea I was thinking this would basically replace/deprecate llama logger.

We could remove llama logger in this PR, or at least mark it as deprecated ?

Reply:
Sounds good. Let's deprecate it in a separate PR to keep the scope of this PR low.